### PR TITLE
do not rely on classList for event target checks

### LINKED ISSUE #10133: Rework code to not use class names
## Test plan

https://github.com/brave/browser-laptop/pull/10165#issue-245912630

----

Discovered by @jonathansampson and @kevinlawler while troubleshooting https://github.com/brave/browser-laptop/issues/10029

### Our current theory on the root cause:
The Aphrodite library just introduced a minify option (specifically with https://github.com/Khan/aphrodite/commit/7f9f6028ec9600d8f20f39ac54431d8b801029d9).  Unfortunately, we have checks which rely on the class name and we DO set NODE_ENV = production while doing a build (which triggers this option OR a similar option).

### Example of problem
Here's an example of the issue, showing the difference in the same element between two versions of Brave (notice the `class` attribute is different):
0.17.16:
> `<div class="popupWindow_yfkq6x-o_O-reverseExpand_142fl4c" style="height: 459px; top: 42px; width: 352px; right: 1em;">...</div>`

0.17.17:
> `<div class="_yfkq6x-o_O-_142fl4c" style="height: 459px; top: 44px; width: 352px; right: 1em;">...</div>`

### The proposed fix:
Update the following code to check using attributes, not class names:
https://github.com/brave/browser-laptop/blob/2e92a017151b77d87031fef4e030ced9dd33f00f/app/renderer/components/main/main.js#L481

### PR DESCRIPTION
**Pull request by cezaraugusto**
Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Auditors: @bsclifton, @jonathansampson
Close #10133

Test Plan:
```
npm run test -- --grep="eventUtil"
```

Related: #10062 

Manual test plan (cc @luixxiul):

1. Update Aphrodite to `1.2.3` in package.json
2. `rm -rf node_modules/ && rm -rf ~/.electron && npm i`
3. `NODE_ENV=production npm run watch`
4. `npm start`
5. Open Devtool with Shift+F8.
6. Check that styles for tabs should show the style concatenated
7. You should be able to click extension button and open extensions popup, which should be persistent;
8. Bookmark hanger should work as expected;
9. Right click over any of the above should not close the dialog.

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

**Comment by luixxiul** [comment]
@cezaraugusto to test the effectiveness of this PR, Aphrodite should be updated to `1.2.3`, where we had the issue on the packaged build (because on that version Aphrodite started shrinking the classnames when NODE_ENV = production is set).

**Comment by luixxiul** [comment]
@cezaraugusto @jonathansampson @bsclifton I updated the manual test plan. Would you mind confirming that it works? thanks!

File: js/lib/eventUtil.js [hunk 1]
```diff
@@ -10,6 +10,11 @@ module.exports.isForSecondaryAction = (e) =>
   e.button === 1
 
 module.exports.eventElHasAncestorWithClasses = (e, classesToCheck) => {
+  // DO NOT ADD NEW CHECKS USING THIS METHOD
+  // classNames are changed from dev to prod by Aphrodite
```
Comment (luixxiul) [inline js/lib/eventUtil.js]: @cezaraugusto mind specifying Aphrodite version (`1.2.3`) and adding the number of the issue (#10029)?

**Comment by luixxiul** [comment]
this should fix #10104 as well

File: test/unit/js/lib/eventUtilTest.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+/* global describe, it */
+
+const eventUtil = require('../../../../js/lib/eventUtil')
+const assert = require('assert')
+
+require('../../braveUnit')
+
+describe('eventUtil', function () {
```
Comment (bsclifton) [inline test/unit/js/lib/eventUtilTest.js]: @cezaraugusto can you add a test for a partial match?  like `nespressoPods` and/or `love_nespresso`?

**Comment by bsclifton** [comment]
@luixxiul @cezaraugusto can you confirm this also closes https://github.com/brave/browser-laptop/issues/10104?

File: test/unit/js/lib/eventUtilTest.js (hunk 2 above)
Comment (cezaraugusto) [inline test/unit/js/lib/eventUtilTest.js]: added `can not accept partial string match`

**Comment by luixxiul** [comment]
could I know how to test the binary build with `NODE_ENV=production` for manual QA? See: https://github.com/brave/browser-laptop/pull/10165#issue-245912630

**Comment by cezaraugusto** [comment]
@luixxiul pls run `NODE_ENV=production npm run watch` when running the app. Notice you'll need to shift+f8 to open devtools. Checking styles for tabs should show the style concatenated

**Comment by luixxiul** [comment]
@cezaraugusto I tried `NODE_ENV=production npm run watch` and it did not work for me. I modified `watch` and `start` inside package.json by adding `cross-env NODE_ENV=production` and that did not work either :-( do you have any tip?

**Comment by luixxiul** [comment]
`NODE_ENV=production npm run watch` does not seem to work because of this error:

````
    ERROR in webtorrentPage.entry.js from UglifyJs
    SyntaxError: Name expected [./app/renderer/components/styles/commonStyles.js:5,6]
webpack: Failed to compile.
````

**Pull request merged**